src/wasm-lib/kcl/src/std/assert.rs

```diff
@@ -0,0 +1,121 @@
+//! Standard library assert functions.
+
+use anyhow::Result;
+use derive_docs::stdlib;
+use schemars::JsonSchema;
+
+use crate::{
+    errors::{KclError, KclErrorDetails},
+    executor::MemoryItem,
+    std::Args,
+};
+
+async fn _assert(value: bool, message: &str, args: &Args) -> Result<(), KclError> {
+    if !value {
+        return Err(KclError::Type(KclErrorDetails {
+            message: format!("assert failed: {}", message),
+            source_ranges: vec![args.source_range],
+        }));
+    }
+    Ok(())
+}
+
+/// Check that the provided value is true, or raise a [KclError]
+/// with the provided description.
+pub async fn assert(args: Args) -> Result<MemoryItem, KclError> {
+    let (data, description): (bool, String) = args.get_data()?;
+    args.make_user_val_from_f64(inner_assert(data, &description, &args).await?)
+}
+
+/// Check a value at runtime, and raise an error if the argument provided
+/// is false.
+///
+/// ```no_run
+/// const myVar = true
+/// assert(myVar, "should always be true")
+/// ```
+#[stdlib {
+    name = "assert",
+}]
+async fn inner_assert(data: bool, message: &str, args: &Args) -> Result<f64, KclError> {
+    _assert(data, message, args).await?;
+    Ok(0.0)
```

Comment:
What's the reason for returning 0? I think that if it has to return something, it would be a boolean. But even that is strange because it implies it can be used in a conditional, like:

```javascript
if assertLessThan(x, 100, "x should be less than 100") {
  // Something
}
```

I would probably return `Ok(())` and `assert()` can return a `null` JSON value.

Reply:
I was scanning functions and there's no easily findable precedent that I could find for a null/empty slice return; this seemed like it'd break less. I'm happy to make it return a () but it may break unexpected things if kcl assumes all functions have a return

Reply:
made it nil and i'll let y'all deal with the fallout ;)
![image](https://github.com/user-attachments/assets/e0c0cb8d-5a3e-442a-b91d-f19abdc5ad19)